compiler/rustc_trait_selection/src/traits/mod.rs

```diff
@@ -366,11 +367,19 @@ pub fn normalize_param_env_or_error<'tcx>(
                     if c.has_escaping_bound_vars() {
                         return ty::Const::new_misc_error(self.0);
                     }
+
                     // While it is pretty sus to be evaluating things with an empty param env, it
                     // should actually be okay since without `feature(generic_const_exprs)` the only
                     // const arguments that have a non-empty param env are array repeat counts. These
                     // do not appear in the type system though.
-                    c.normalize_internal(self.0, ty::ParamEnv::empty())
+                    if let ty::ConstKind::Unevaluated(uv) = c.kind()
```

Comment:
I wonder if instead of making a new infcx, we should possibly instead move this folder into `do_normalize_predicates`, where we do make a new infcx. I guess we expect that this just evaluates to a valtree or rigid const?

If not, we should at least assert that nothing is leaking out of this infcx (infer vars/placeholders).

Reply:
`do_normalize_predicates` doesn't work as we attempt to normalize the predicates in an environment that has unnormalized anon consts. I will update this to assert we dont have any infcx local state leak out though I think this is pretty unfortunate...